src/include/machnet_engine.h

```diff
@@ -874,7 +962,73 @@ class MachnetEngine {
         if (listeners_.find(local_ipv4_addr) != listeners_.end()) {
           // We have a listener on this port.
           const auto &listeners_on_ip = listeners_[local_ipv4_addr];
+          const int qid =
+              shared_state_->GetQueueId(local_ipv4_addr, local_udp_port);
           if (listeners_on_ip.find(local_udp_port) == listeners_on_ip.end()) {
+            if (qid > -1) {
+              auto *response = CHECK_NOTNULL(packet_pool_->PacketAlloc());
+              const size_t hdr_length =
+                  (sizeof(Ethernet) + sizeof(Ipv4) + sizeof(Udp) +
+                   sizeof(net::MachnetPktHdr));
+              const uint32_t pkt_len =
+                  hdr_length +
+                  46;  // I am assuming the rss key size to 40 bytes.
+              Ethernet *response_eh = response->append<Ethernet *>(pkt_len);
```

Comment:
You can use `auto *` here. You can do:

`auto *response_eh = CHECK_NOTNULL(response->append<Ethernet >(pkt_len));`